distribution/lib/Standard/Table/0.0.0-dev/src/Table.enso

```diff
@@ -673,10 +675,20 @@ type Table
               table.rename_columns (Map.from_vector [["name=(.*)".to_regex, "key:$1"]])
     @column_map Widget_Helpers.make_rename_name_vector_selector
     rename_columns : Map (Text | Integer | Regex) Text | Vector Text | Vector Vector -> Case_Sensitivity -> Boolean -> Problem_Behavior -> Table ! Missing_Input_Columns | Ambiguous_Column_Rename | Too_Many_Column_Names_Provided | Invalid_Column_Names | Duplicate_Output_Column_Names
-    rename_columns self (column_map:(Map | Vector)=["Column"]) (case_sensitivity:Case_Sensitivity=Case_Sensitivity.Default) (error_on_missing_columns:Boolean=True) (on_problems:Problem_Behavior=Report_Warning) =
-        new_names = Table_Helpers.rename_columns self.column_naming_helper self.columns column_map case_sensitivity error_on_missing_columns on_problems
-        Warning.with_suspended new_names names->
-            Table.new (self.columns.map c-> c.rename (names.at c.name))
+    rename_columns self (column_map:(Table | Map | Vector)=["Column"]) (case_sensitivity:Case_Sensitivity=Case_Sensitivity.Default) (error_on_missing_columns:Boolean=True) (on_problems:Problem_Behavior=Report_Warning) = case column_map of
+        _ : Table -> case column_map.column_count of
+            1 ->
+                col = column_map.first_column
+                if col.value_type.is_text then self.rename_columns col.to_vector case_sensitivity error_on_missing_columns on_problems else
+                    Error.throw (Illegal_Argument.Error "Expected a table with one or two columns of text values.")
+            2 ->
+                if column_map.first_column.value_type.is_text && column_map.second_column.value_type.is_text then self.rename_columns (Map.from_vector column_map.rows) case_sensitivity error_on_missing_columns on_problems else
+                    Error.throw (Illegal_Argument.Error "Expected a table with one or two columns of text values.")
+            _ -> Error.throw (Illegal_Argument.Error "Expected a table with one or two columns of text values.")
+        _ ->
```

Comment:
This error checking seems to be duplicated across `Table` and `DB_Table`.

Can we move it to `Table_Helpers` to keep common code and DRY?